### PR TITLE
Make liveness and readiness probes configurable

### LINKED ISSUE #19623: [stable/rocketchat] Liveness and Readiness probes should be configurable
**Is your feature request related to a problem? Please describe.**

On some slower systems, it takes too long for Rocketchat to reach a state in which it responds to the liveness probe. If the installer can configure the probe, this problem can be addressed by the installer.

**Describe the solution you'd like**

Configurable probes, like in the Nextcloud chart, for example

**Describe alternatives you've considered**

If I can't configure the probe I'd have to kustomize it or fork the chart :thumbsdown:

**Additional context**

No additional context. I'll make a PR soon

### PR DESCRIPTION
**Pull request by mrtndwrd**
#### What this PR does / why we need it:

Closes #19623 

#### Which issue this PR fixes

  - fixes # #19623 

#### Special notes for your reviewer:

Tagging the maintainers so this can be merged quickly

@geekgonecrazy @pierreozoux @verwilst @srueg 

#### Checklist

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/rocketchat/README.md [hunk 1]
```diff
@@ -75,6 +75,18 @@ Parameter | Description | Default
 `ingress.annotations` | Annotations for the ingress | `{}`
 `ingress.path` | Path of the ingress | `/`
 `ingress.tls` | A list of [IngressTLS](https://kubernetes.io/docs/reference/federation/extensions/v1beta1/definitions/#_v1beta1_ingresstls) items | `[]`
+`livenessProbe.enabled` | Turn on and off liveness probe | `true`
+`livenessProbe.initialDelaySeconds` | Delay before liveness probe is initiated | `30`
```
Comment (srueg) [inline stable/rocketchat/README.md]:
```suggestion
`livenessProbe.initialDelaySeconds` | Delay before liveness probe is initiated | `60`
```

File: stable/rocketchat/README.md [hunk 2]
```diff
@@ -75,6 +75,18 @@ Parameter | Description | Default
...
+`livenessProbe.periodSeconds` | How often to perform the probe | `15`
+`livenessProbe.timeoutSeconds` | When the probe times out | `5`
+`livenessProbe.failureThreshold` | Minimum consecutive failures for the probe | `3`
+`livenessProbe.successThreshold` | Minimum consecutive successes for the probe | `1`
+`readinessProbe.enabled` | Turn on and off readiness probe | `true`
+`readinessProbe.initialDelaySeconds` | Delay before readiness probe is initiated | `30`
```
Comment (srueg) [inline stable/rocketchat/README.md]:
```suggestion
`readinessProbe.initialDelaySeconds` | Delay before readiness probe is initiated | `10`
```

**Review by srueg — CHANGES_REQUESTED** [review]

**Comment by mrtndwrd** [comment]
Thanks for the suggestions! I amended the previous commit with the right README values

**Comment by mrtndwrd** [comment]
`pull-charts-e2e` failed. What does this test do? Is it related to my changes?

**Comment by mrtndwrd** [comment]
@srueg it seems like these test failures are not related to the rocketchat chart? If I understand the logs correctly it starts the pods correctly, everything's fine, then something weird happens..? I don't think I have access to the test source, so I can't really see what causes this python error: 

```
...
 I1217 14:38:09.624] ========================================================================================================================
I1217 14:38:09.624] Deleting release 'rocketchat-q4ipib8tcc'...
I1217 14:38:10.680] release "rocketchat-q4ipib8tcc" deleted
I1217 14:38:10.684] Deleting namespace 'rocketchat-presubmit-19637-1206943879878676480-q4ipib8tcc'...
I1217 14:38:10.983] namespace "rocketchat-presubmit-19637-1206943879878676480-q4ipib8tcc" deleted
I1217 14:38:49.895] Namespace 'rocketchat-presubmit-19637-1206943879878676480-q4ipib8tcc' terminated.
I1217 14:38:49.896] Error linting and installing charts: Error processing charts
I1217 14:38:49.896] ------------------------------------------------------------------------------------------------------------------------
I1217 14:38:49.896]  ✖︎ rocketchat => (version: "2.0.1", path: "stable/rocketchat") > Error waiting for process: exit status 1
I1217 14:38:49.896] ------------------------------------------------------------------------------------------------------------------------
I1217 14:38:50.357] b682dc5dabd91826a24abc9849467f3bd353d0e4001e9e8db00a048e5e689d5f
W1217 14:38:50.370] Traceback (most recent call last):
W1217 14:38:50.370]   File "/workspace/./test-infra/jenkins/../scenarios/execute.py", line 50, in <module>
W1217 14:38:50.370]     main(ARGS.env, ARGS.cmd + ARGS.args)
W1217 14:38:50.370]   File "/workspace/./test-infra/jenkins/../scenarios/execute.py", line 41, in main
W1217 14:38:50.371]     check(*cmd)
W1217 14:38:50.371]   File "/workspace/./test-infra/jenkins/../scenarios/execute.py", line 30, in check
W1217 14:38:50.371]     subprocess.check_call(cmd)
W1217 14:38:50.371]   File "/usr/lib/python2.7/subprocess.py", line 190, in check_call
W1217 14:38:50.371]     raise CalledProcessError(retcode, cmd)
W1217 14:38:50.371] subprocess.CalledProcessError: Command '('./test/e2e.sh',)' returned non-zero exit status 1 
```

**Comment by mrtndwrd** [comment]
@srueg @davidkarlsen is there anything I can do to speed up this process? I need this change for my project and I don't like having to use my own fork to install the chart. If you can point me to why the test is failing I'd be happy to fix it


/retest

**Comment by srueg** [comment]
I also don't know the test setup that well and can't see the obvious error. I'll take another look.

**Comment by srueg** [comment]
I opened another PR without any changes to test the CI process: https://github.com/helm/charts/pull/19696 which also failed.

You can find the logs here: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/helm_charts/19637/pull-charts-e2e/1207613472548851712/

It seems that the Rocke.Chat container crashes for some reason. Unfortunately the logs only show the current pod which didn't fail (or didn't log anything).

**Comment by srueg** [comment]
Maybe @unguiculus could give us a hand here?

**Review by pierreozoux — APPROVED** [review]
Lgtm sorry for delay

**Comment by mrtndwrd** [comment]
Hi, this MR has been approved by 1 reviewer. Some tests still fail, but that shouldn't be due to my changes. Do you want me to merge master into this branch to see if that fixes the problem? Or can it otherwise be approved by a second reviewer? Thanks!

**Comment by srueg** [comment]
Yes, let's try to merge (or rebase) master. The tests are required to succeed in order to merge it...

**Comment by mrtndwrd** [comment]
```
I0224 13:28:26.352] Installing chart with values file 'stable/rocketchat/ci/ci-values.yaml'...
I0224 13:28:26.352] 
I0224 13:38:28.077] Error: release rocketchat-o311slu9hb failed: timed out waiting for the condition
I0224 13:38:28.078] ========================================================================================================================
I0224 13:38:28.078] ........................................................................................................................
I0224 13:38:28.079] ==> Events of namespace rocketchat-presubmit-19637-1231933663680139264-o311slu9hb
I0224 13:38:28.079] ........................................................................................................................ 
```

Should I increase the timeout in `values.yaml`?

**Comment by srueg** [comment]
No idea why this fails. In the logs it looks like RocketChat successfully starts up...
Which timeout would you increase? You could also change this in `ci/ci-values.yaml` if it's specific to CI

**Comment by mrtndwrd** [comment]
/retest

Hmm, I'm not sure which timeout it runs into exactly, TBH. On slow systems, we have seen that pods get restarted because they don't react to the liveness probe in time, so we've increased the `livenessProbe.initialDelaySeconds` for other projects.

We can also try to retest to see if it's a consistent problem

**Comment by mrtndwrd** [comment]
Yay, test finally succeeded!

**Pull request merged**